src/world.js

```diff
@@ -41,27 +34,27 @@ class World extends EventEmitter {
       return true
     }
     const ps = []
-    const chunkLength = Math.ceil((length + modulus(iniPos.z, 16)) / 16)
-    const chunkWidth = Math.ceil((width + modulus(iniPos.x, 16)) / 16)
+    const iniPosInChunk = posInChunk(iniPos)
+    const chunkLength = Math.ceil((length + iniPosInChunk.z) / 16)
+    const chunkWidth = Math.ceil((width + iniPosInChunk.x) / 16)
     for (let chunkZ = 0; chunkZ < chunkLength; chunkZ++) {
       const actualChunkZ = chunkZ + Math.floor(iniPos.z / 16)
       for (let chunkX = 0; chunkX < chunkWidth; chunkX++) {
         const actualChunkX = chunkX + Math.floor(iniPos.x / 16)
         ps.push(this.getColumn(actualChunkX, actualChunkZ)
           .then(chunk => {
-            const offsetX = chunkX * 16 - modulus(iniPos.x, 16)
-            const offsetZ = chunkZ * 16 - modulus(iniPos.z, 16)
+            const offsetX = chunkX * 16 - iniPosInChunk.x
+            const offsetZ = chunkZ * 16 - iniPosInChunk.z
             chunk.initialize((x, y, z) => inZone(x + offsetX, y - iniPos.y, z + offsetZ) ? iniFunc(x + offsetX, y - iniPos.y, z + offsetZ) : null)
             return this.setColumn(actualChunkX, actualChunkZ, chunk)
           })
-          .then(() => ({chunkX: actualChunkX, chunkZ: actualChunkZ})))
+          .then(() => ({ chunkX: actualChunkX, chunkZ: actualChunkZ })))
       }
     }
     return Promise.all(ps)
   };
 
   async getColumn (chunkX, chunkZ) {
-    await Promise.resolve()
```

Comment:
This line does nothing yes but it's important. Without it people may start assuming the API is sync in some cases which we definitely don't want.

Reply:
ok